audit.tf

```diff
@@ -87,6 +87,48 @@ resource "aws_sns_topic_policy" "monitor_iam_access_audit" {
   policy   = data.aws_iam_policy_document.monitor_iam_access_audit_topic.json
 }
 
+resource "aws_guardduty_detector" "audit" {
+  count    = var.aws_guardduty == true ? 1 : 0
+  provider = aws.audit
+}
+
+resource "aws_guardduty_member" "logging" {
+  count       = var.aws_guardduty == true ? 1 : 0
+  provider    = aws.audit
+  account_id  = aws_guardduty_detector.logging[0].account_id
+  detector_id = aws_guardduty_detector.audit[0].id
+  email       = local.aws_account_emails[aws_guardduty_detector.logging[0].account_id]
+  invite      = true
+
+  depends_on = [aws_guardduty_organization_admin_account.audit]
+  lifecycle {
+    ignore_changes = [email]
```

Comment:
In case anyone wonders why we have `ignore_changes = [email]` here, it's because we are enabling GuardDuty's member accounts via AWS Organizations and AWS doesn't return the email in the API response like in the example below.

```
aws guardduty list-members --detector-id 123abc
{
    "Members": [
        {
            "AccountId": "123456789012",
            "DetectorId": "60b",
            "MasterId": "123456789011",
            "RelationshipStatus": "Enabled",
            "UpdatedAt": "..."
        }
    ]
```

This behaviour triggers Terraform to recreate the resource and add the missing email value.